Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformer.java

```diff
@@ -35,45 +28,60 @@
 public class NetworkTransformer implements INetworkTransformer {
 
     @Autowired
-    private INetworkManager networkManager;
-
-    @Autowired
-    private IConceptCollectionManager conceptCollectionManager;
+    private EventParser parser;
 
     @Autowired
-    private EventParser parser;
+    private ElementEventTypeDownloadService elementEventTypeDownloadService;
 
     private static final Logger logger = LoggerFactory
             .getLogger(NetworkTransformer.class);
 
+    private static int NUM_THREADS = 4;
+
     /**
      * 
      * {@inheritDoc}
      */
     @Override
     public ITransformedNetwork transformNetwork(
             List<INetworkNodeInfo> networkTopNodesList) {
-        Map<String, Node> nodes = new HashMap<String, Node>();
-        List<Link> links = new ArrayList<Link>();
+        Map<String, Node> nodes = new HashMap<>();
+        List<Link> links = new ArrayList<>();
         ITransformedNetwork transformedNetwork = new TransformedNetwork(nodes, links);
 
-        if (networkTopNodesList != null && networkTopNodesList.size() > 0) {
-            for (INetworkNodeInfo networkNodeInfo : networkTopNodesList) {
-                if (networkNodeInfo.getStatementType().equals(
-                        INetworkManager.RELATIONEVENT)) {
-                    try {
-                        parser.parseStatement(networkNodeInfo.getId(),
-                                nodes, links);
-                    } catch (JAXBException e) {
-                        logger.error("Issue while parsing the JAXB object",
-                                e);
-                    } catch (QStoreStorageException e) {
-                        logger.error("QStore retrieve error", e);
-                    }
-                }
-            }
+        if (networkTopNodesList == null || networkTopNodesList.size() == 0) {
+            // return the trasnformed network
+            return transformedNetwork;
         }
 
+        // networkTopNode list has size > 0
+        // Create executors and tasks
+        ExecutorService executorService = Executors.newFixedThreadPool(NUM_THREADS);

```

Comment:
i'm pretty sure you could have used  your previous code, as the taskExecutor implemented the same interface as the executorservice. But looks fine too.